frontend/src/metabase/query_builder/components/filters/modals/InlineCategoryPicker/InlineCategoryPicker.unit.spec.tsx

```diff
@@ -354,14 +359,49 @@ describe("InlineCategoryPicker", () => {
         filter={testFilter}
         newFilter={testFilter}
         onChange={changeSpy}
-        fieldValues={smallCategoryField.values}
+        fieldValues={largeCategoryField.values}
         fetchFieldValues={fetchSpy}
-        dimension={smallDimension}
+        dimension={largeDimension}
         onClear={changeSpy}
       />,
     );
 
     expect(screen.queryByTestId("category-picker")).not.toBeInTheDocument();
     expect(screen.queryByTestId("select-button")).toBeInTheDocument();
   });
+
+  const fieldSizes = [
+    { name: "large", field: largeCategoryField, dimension: largeDimension },
+    { name: "small", field: smallCategoryField, dimension: smallDimension },
+  ];
+
+  fieldSizes.forEach(({ name, field, dimension }) => {
+    it(`should fall back to value picker if the filter operator is not = or != with a ${name} set of field values`, () => {
+      const testFilter = new Filter(
+        ["contains", ["field", field.id, null], undefined],
+        null,
+        query,
+      );
+      const changeSpy = jest.fn();
+      const fetchSpy = jest.fn();
+
+      render(
+        <Provider store={store}>
```

Comment:
What do you think about using `renderWithProviders`, which seems more established instead? https://github.com/metabase/metabase/blob/43f084bf3cbca42abdd78ed9c78c58fcda380dec/frontend/test/__support__/ui.js#L42

Reply:
this is lovely! I didn't know about it!